### ISSUE TITLE
Attention visualization: cvpr2021 paper vs code mismatch

### ISSUE THREAD

**Issue by pimpke** [autonomousvision/transfuser#131]
Hi, I'm running the code from the cvpr2021 branch.

I've found the exact frame from the validation dataset you used for producing the below image in the 2021 transfuser paper showing attention:
![image](https://user-images.githubusercontent.com/7732949/207388478-eb9dde56-b63b-4d8b-aa4c-595b6e9ccbf8.png)
(Figure 3. in the [paper](https://arxiv.org/pdf/2104.09224.pdf#page=8))

Here are the attention visualizations I got when I ran viz.py on that frame (The following images are for the first 3 heads and the 4th head doesn't produce attention visualization for the "yellow" traffic light image token). The first image in a row is showing the query image patch token, and the second and the third are showing the top 5 attended image/lidar tokens.
![22](https://user-images.githubusercontent.com/7732949/207389141-73f1f9ff-c9fc-47bc-b643-b72ebeaeaef6.png)
![22](https://user-images.githubusercontent.com/7732949/207389163-038ffe99-284e-4e50-85b9-dd0c09e7739d.png)
![22](https://user-images.githubusercontent.com/7732949/207389193-0bd3ccf1-a9bb-4c3e-9ea1-bec357847404.png)
(Please ignore the color channel swap in the above images)

The transfuser model I used in viz.py was downloaded from the link in cvpr2021's README (https://s3.eu-central-1.amazonaws.com/avg-projects/transfuser/models.zip)

Do you know why I get this mismatch in attention visualization when I run it locally compared to the paper's attention visualization?

Thank you!

**Comment by ap229997** [comment]
We had updated the checkpoint with the model that was submitted to the official CARLA leaderboard. This model is trained on a different dataset than the one used for results and visualizations in the paper. I'd suggest shifting to the 2022 branch since the models are much better.

**Comment by Kait0** [comment]
We fixed some bugs at some point that required retraining the model (the image normalization values were wrong and we added action repeat of 2 to avoid a problem with the LiDAR Hz being different than the camera).
The model in the github repo is not the exact same one anymore than in the CVPR paper.
It is just some example weight at this point, it was also trained with a different expert driver / dataset.
I suppose that is where the difference comes from. Different training runs will converge to different attention behaviors.
